server/routerlicious/packages/services-shared/src/healthCheckEndpoints.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Router } from "express";
+import { StartupChecker } from "./startupChecker";
+import { LumberEventName, Lumberjack } from "@fluidframework/server-services-telemetry";
+
+/**
+ * Checks if a service or functionality is ready for use.
+ * @internal
+ */
+export interface IReadinessCheck {
+	/**
+	 * Whether the service/functionality is ready for use.
+	 */
+	isReady(): Promise<boolean>;
+}
+
+/**
+ * Creates the health check endpoints for the service.
+ * @param serviceName - The name of the service.
+ * @param readinessCheck - Optional readiness check.
+ * @param createLivenessEndpoint - Whether to create the liveness endpoint. Services like Alfred already have a ping endpoint, so this can be set to false.
+ */
+export function createHealthCheckEndpoints(
+	serviceName: string,
+	readinessCheck?: IReadinessCheck,
+	createLivenessEndpoint = true,
```

Comment:
Did this to reduce the number of args passed to the `createHealthCheckEndpoints`. For example, `IReadiness` is passed as an arg from all services while `createLivenessEndpoint` is set to `false` only in `Alfred.

Example for Nexus and Alfred:

`const healthEndpoints = createHealthCheckEndpoints("nexus", readinessCheck);`
`const healthCheckEndpoints = createHealthCheckEndpoints(
		"alfred",
		readinessCheck,
		false /* createLivenessEndpoint */,
	);`

If I move the parameter for `IReadiness` behind the `createLivenessEndpoint`, it will make the code more verbose